pkg/controller/backupbucket/backupbucket.go

```diff
@@ -6,19 +6,27 @@ package backupbucket
 
 import (
 	"context"
+	"crypto/sha1"
+	"encoding/hex"
 	"fmt"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
-	"github.com/gardener/gardener/pkg/utils"
 
 	azureclient "github.com/gardener/gardener-extension-provider-azure/pkg/azure/client"
 )
 
+// toSHA1 takes a byte slice and returns the sha1-hashed byte slice.
+func toSHA1(in []byte) []byte {
+	s := sha1.New()
+	_, _ = s.Write(in)
+	return s.Sum(nil)
+}
+
 func ensureBackupBucket(ctx context.Context, factory azureclient.Factory, backupBucket *extensionsv1alpha1.BackupBucket) (string, string, error) {
 	var (
-		backupBucketNameSha = utils.ComputeSHA1Hex([]byte(backupBucket.Name))
+		backupBucketNameSha = hex.EncodeToString(toSHA1([]byte(backupBucket.Name)))
```

Comment:
Do we need to preserve the old bucket hash at all? Because the `ensureBackupBucket` method with a new hash seems to be only called if no prior backup bucket (secret) exists:

https://github.com/gardener/gardener-extension-provider-azure/blob/d7f9b49272c3f6a8acd007f4b3e23d955a1c1ca3/pkg/controller/backupbucket/actuator.go#L62-L63